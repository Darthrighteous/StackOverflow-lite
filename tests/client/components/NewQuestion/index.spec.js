import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import ConnectedNewQuestion, { NewQuestion } from '../../../../client/components/NewQuestion';

const mockStore = configureMockStore([]);
const store = mockStore({});

/**
 * Function to set up
 * @returns {object} - object containing wrapper
 */
function setup() {
  const props = {
    postQuestion: jest.fn(),
    history: {
      push: jest.fn(),
      goBack: jest.fn(),
    },
  };

  const connectedWrapper = shallow(
    <ConnectedNewQuestion {...props} store={store} />
  );
  const newQuestionWrapper = shallow(<NewQuestion {...props} />);
  return {
    connectedWrapper,
    newQuestionWrapper
  };
}

describe('New Question snapshot test', () => {
  const { connectedWrapper, newQuestionWrapper } = setup();
  test('snapshot', () => {
    expect(connectedWrapper).toMatchSnapshot();
    expect(newQuestionWrapper).toMatchSnapshot();
  });
});

describe('event simulations', () => {
  const { newQuestionWrapper } = setup();
  it('should handle input change', () => {
    const event = {
      target: { name: 'title', value: 'testtitle' }
    };

    newQuestionWrapper.find('input').first().simulate('change', event);
    expect(newQuestionWrapper.state().title).toEqual('testtitle');
    
  });

  it('should handle submit', () => {
    const event = {
      preventDefault: jest.fn()
    };
    newQuestionWrapper.find('.new-question-form').simulate('submit', event);
    expect(event.preventDefault).toHaveBeenCalled();
  });

  it('should handle cancel', () => {
    newQuestionWrapper.find('.btn-cancel').simulate('click');
    expect(newQuestionWrapper.state().title).toEqual('');
    expect(newQuestionWrapper.state().body).toEqual('');


    newQuestionWrapper.find('.btn-cancel').simulate('click');
  });
});
