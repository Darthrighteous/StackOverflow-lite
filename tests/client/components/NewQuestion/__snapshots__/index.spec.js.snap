// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Question snapshot test snapshot 1`] = `
<NewQuestion
  history={
    Object {
      "goBack": [MockFunction],
      "push": [MockFunction],
    }
  }
  postQuestion={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`New Question snapshot test snapshot 2`] = `
<div
  className="main-content"
>
  <div
    className="content-header"
  >
    <h2>
      New Question
    </h2>
  </div>
  <form
    className="new-question-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="title"
    >
      <span>
        Title
      </span>
      <input
        autoComplete="off"
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="add title here"
        required={true}
        value=""
      />
    </label>
    <label
      htmlFor="body"
    >
      <span>
        Body
      </span>
      <textarea
        autoComplete="off"
        id="body"
        name="body"
        onChange={[Function]}
        placeholder="add body here"
        required={true}
        value=""
      />
    </label>
    <div
      className="save-options"
    >
      <button
        className="btn-cancel"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
      <button
        className="btn-post"
        type="submit"
      >
        Post
      </button>
    </div>
  </form>
</div>
`;
